meerk40t/gui/wxmribbon.py

```diff
@@ -103,14 +110,113 @@ def GetBestSize(self):
                     ht += 10
                     if self.GetLabel() != "":
                         ht += 23
-                    # print ("Found: wd=%s, ht=%s" % (wd, ht))
-                    size = wx.Size(wd, ht)
+                    else:
+                        ht += 10
                     break
+        else:
+            wd = size[0]
+            ht = size[1]
+        if platform.system != "Windows":
+            ht = max(ht, 120)
+            maxw = 0
+            for bar in self.GetChildren():
+                ct = 0
+
+                if isinstance(bar, RB.RibbonButtonBar):
+                    for button in bar._buttons:
+                        w, h = button.bitmap_large.GetSize()
+                        maxw = max(maxw, w + 10)
+                        ct += 1
+                    maxw = max(maxw, 25 + 10)
+            # print ("Ct=%d, widest=%d, wd=%.1f, wd2=%.1f, ht=%.1f, oldh=%.1f" % (ct, maxw, wd, 1.5*ct*maxw, ht, oldh))
+            wd = max(wd, 1.5 * ct * maxw)
+        size = wx.Size(wd, ht)
+        # print (size, size2)
+        # size = size2
         return size
 
 
+    def IsMinimised(self, at_size=None):
+        # Very much simplified version..
+        if self.recurse:
+            res = False
+        else:
+            res = super().IsMinimised(at_size)
+        return res
+
+    def ShowExpanded(self):
+        """
+        Show the panel externally expanded.
+
+        When a panel is minimised, it can be shown full-size in a pop-out window, which
+        is referred to as being (externally) expanded.
+
+        :returns: ``True`` if the panel was expanded, ``False`` if it was not (possibly
+         due to it not being minimised, or already being expanded).
+
+        :note: When a panel is expanded, there exist two panels - the original panel
+         (which is referred to as the dummy panel) and the expanded panel. The original
+         is termed a dummy as it sits in the ribbon bar doing nothing, while the expanded
+         panel holds the panel children.
+
+        :see: :meth:`~RibbonPanel.HideExpanded`, :meth:`~RibbonPanel.GetExpandedPanel`
+        """
+
+        if not self.IsMinimised():
+            return False
+
+        if self._expanded_dummy is not None or self._expanded_panel != None:
```

Comment:
Where do these come from and do they always exist? Codacy insists the property isn't set yet, and I can't find anything to contradict that.

Reply:
It's inheriting the agw class and uses those properties. Codacy isn't aware of that source I think